vermouth/rcsu/get_go.py

```diff
@@ -0,0 +1,164 @@
+
+def _get_go_type(molecule, resid, chain, prefix):
+    for node in molecule.nodes:
+        attrs = molecule.nodes[node]
+        if attrs['resid'] == resid and attrs['chain'] == chain:
+            if prefix in attrs['atype']:
+                return attrs['atype']
+    raise IOError(f"Could not find GoVs with resid {resid} in chain {chain}.")
+
+def _in_region(resid, regions):
+    """
+    Check if resid falls in regions interval.
+    """
+    for low, up in regions:
+        if low <= resid <= up:
+            return True
+    return False
+
+class GetGo():
+    """
+    Generate the Go model interaction parameters.
+    """
+    def __init__(self,
+                 contact_map,
+                 cutoff_short,
+                 cutoff_long,
+                 go_eps,
+                 res_dist,
+                 res_graph,
+                 regions=None,
+                 water_bias=None,
+                 idp_water_bias=None,
+                 prefix="VS"):
+
+        self.contact_map = contact_map
+        self.cutoff_short = cutoff_short
+        self.cutoff_long = cutoff_long
+        self.go_eps = go_eps
+        self.res_dist = res_dist
+        self.itp_file = open('GoVirtIncludes.itp', "w", encoding='UTF-8')
+        self.prefix = prefix
+        self.res_graph = res_graph
+        self.water_bias = water_bias
+        self.idp_water_bias = idp_water_bias
+        self.regions = regions
+
+    def write_go_structural_bias(self, molecule):
+        chain_id_to_resnode = {}
+        # find all pairs of residues that are within bonded distance of
+        # self.res_dist
+        connected_pairs = dict(nx.all_pairs_shortest_path_length(self.res_graph,
+                                                            cutoff=self.res_dist))
+
+        # for each residue collect the chain and residue in a dict
+        # we use this later for identifying the residues from the
+        # contact map
+        for resnode in self.res_graph.nodes:
+            chain = self.res_graph.nodes[resnode].get('chain', None)
+            # in vermouth within a molecule all resid are unique
+            # when merging multiple chains we store the old resid
+            # the go model always references the input resid i.e.
+            # the _old_resid
+            resid = self.res_graph.nodes[resnode].get('_old_resid')
+            chain_id_to_resnode[(chain, resid)] = resnode
+
+        for contact in self.contact_map:
+            resIDA, chainA, resIDB, chainB = contact
+            # identify the contact in the residue graph based on
+            # chain ID and resid
+            resA = chain_id_to_resnode[(chainA, resIDA)]
+            resB = chain_id_to_resnode[(chainB, resIDB)]
+            # make sure that both residues are not connected
+            # note: contacts should be symmteric so we should be
+            # able to get rid of the second if clause
+            if resB not in connected_pairs[resA] and resA not in connected_pairs[resB]:
```

Comment:
Yes. Either get rid of it or turn it into an `assert`

Reply:
On second thought, we have had the idea to turn resgraphs into directed graphs. How would that affect this?

Reply:
let's axe the second clause and make a test that checks this; if we decide to change the behavior the test will tell us